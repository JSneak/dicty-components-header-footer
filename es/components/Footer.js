function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import { FooterStyles as styles } from "../styles/FooterStyles";

/**
 * The `Footer` component that will be displayed
 * at the bottom of every React web application of [dictyBase](http://dictybase.org).
 */

var Footer = function (_Component) {
  _inherits(Footer, _Component);

  function Footer() {
    var _temp, _this, _ret;

    _classCallCheck(this, Footer);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.footerItems = function (items) {
      var classes = _this.props.classes;

      return items.map(function (c, i) {
        return React.createElement(
          "li",
          { key: i, className: classes.listItem },
          React.createElement(
            "a",
            { href: c.link, className: classes.link },
            c.description
          )
        );
      });
    }, _this.footerSubSections = function (items) {
      var classes = _this.props.classes;

      return items.map(function (c, i) {
        return React.createElement(
          "div",
          { key: i, className: classes.subsectionItem },
          React.createElement(
            "div",
            null,
            React.createElement(
              "ul",
              { className: classes.headerLink },
              React.createElement(
                "div",
                { className: classes.ulHeader },
                c.header.description
              )
            )
          ),
          React.createElement(
            "ul",
            { className: classes.listItem },
            _this.footerItems(c.items)
          )
        );
      });
    }, _this.footerSections = function (items) {
      var classes = _this.props.classes;

      return items.map(function (c, i) {
        return React.createElement(
          Grid,
          { item: true, key: i, className: classes.gridItem },
          _this.footerSubSections(c)
        );
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Footer.prototype.render = function render() {
    var _props = this.props,
        classes = _props.classes,
        items = _props.items;

    return React.createElement(
      Grid,
      { container: true, className: classes.root, justify: "center" },
      this.footerSections(items)
    );
  };

  return Footer;
}(Component);

export default withStyles(styles)(Footer);