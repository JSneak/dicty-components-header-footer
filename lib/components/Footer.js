"use strict";

exports.__esModule = true;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _styles = require("@material-ui/core/styles");

var _Grid = require("@material-ui/core/Grid");

var _Grid2 = _interopRequireDefault(_Grid);

var _footer = require("../data/footer");

var _footer2 = _interopRequireDefault(_footer);

var _FooterStyles = require("../styles/FooterStyles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * The `Footer` component that will be displayed
 * at the bottom of every React web application of [dictyBase](http://dictybase.org).
 */

var Footer = function (_Component) {
  _inherits(Footer, _Component);

  function Footer() {
    var _temp, _this, _ret;

    _classCallCheck(this, Footer);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.footerItems = function (items) {
      var classes = _this.props.classes;

      return items.map(function (c, i) {
        return _react2.default.createElement(
          "li",
          { key: i, className: classes.listItem },
          _react2.default.createElement(
            "a",
            { href: c.link, className: classes.link },
            c.description
          )
        );
      });
    }, _this.footerSubSections = function (items) {
      var classes = _this.props.classes;

      return items.map(function (c, i) {
        return _react2.default.createElement(
          "div",
          { key: i, className: classes.subsectionItem },
          _react2.default.createElement(
            "div",
            null,
            _react2.default.createElement(
              "ul",
              { className: classes.headerLink },
              _react2.default.createElement(
                "a",
                { href: c.header.link, className: classes.ulLink },
                c.header.description
              )
            )
          ),
          _react2.default.createElement(
            "ul",
            null,
            _this.footerItems(c.items)
          )
        );
      });
    }, _this.footerSections = function (items) {
      var classes = _this.props.classes;

      return items.map(function (c, i) {
        return _react2.default.createElement(
          _Grid2.default,
          { item: true, key: i, className: classes.gridItem },
          _this.footerSubSections(c)
        );
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Footer.prototype.render = function render() {
    var classes = this.props.classes;

    return _react2.default.createElement(
      _Grid2.default,
      { container: true, className: classes.root, justify: "center" },
      this.footerSections(_footer2.default)
    );
  };

  return Footer;
}(_react.Component);

exports.default = (0, _styles.withStyles)(_FooterStyles.FooterStyles)(Footer);
module.exports = exports["default"];