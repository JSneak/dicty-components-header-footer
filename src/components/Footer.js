// @flow
import React, { Component } from "react"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import items from "../data/footer"
import { FooterStyles as styles } from "../styles/FooterStyles"

type ItemType = {
  /** url link */
  link: string,
  /** description of the link that will be displayed */
  description: string,
}

type FooterItemType = {
  /** The header or title of every footer section */
  header: ItemType,
  /** List of entry under the header */
  items: Array<ItemType>,
}

type FooterProps = {
  /** List of footer items, inside a nested list */
  items: Array<Array<FooterItemType>>,
  /** Material-UI's classes object */
  classes: Object,
}

/**
 * The `Footer` component that will be displayed
 * at the bottom of every React web application of [dictyBase](http://dictybase.org).
 */

class Footer extends Component<FooterProps> {
  footerItems = (items: Array<ItemType>) => {
    const { classes } = this.props
    return items.map((c, i) => (
      <li key={i} className={classes.listItem}>
        <a href={c.link} className={classes.link}>
          {c.description}
        </a>
      </li>
    ))
  }

  footerSubSections = (items: Array<FooterItemType>) => {
    const { classes } = this.props
    return items.map((c, i) => (
      <div key={i} className={classes.subsectionItem}>
        <div>
          <ul className={classes.headerLink}>
            <a href={c.header.link} className={classes.ulLink}>
              {c.header.description}
            </a>
          </ul>
        </div>
        <ul>{this.footerItems(c.items)}</ul>
      </div>
    ))
  }

  footerSections = (items: Array<Array<FooterItemType>>) => {
    const { classes } = this.props
    return items.map((c, i) => (
      <Grid item key={i} className={classes.gridItem}>
        {this.footerSubSections(c)}
      </Grid>
    ))
  }
  render() {
    const { classes } = this.props
    return (
      <Grid container className={classes.root} justify="center">
        {this.footerSections(items)}
      </Grid>
    )
  }
}

export default withStyles(styles)(Footer)
