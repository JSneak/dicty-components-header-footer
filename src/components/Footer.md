### Default condensed footer

```jsx
const links = [
  {
    url: "/techniques",
    description: "Techniques",
  },
  {
    url: "/teaching-protocols",
    description: "Teaching Protocols",
  },
  {
    url: "/stockcenter",
    description: "Dicty Stock Center",
  },
  {
    url: "/gene",
    description: "Genome Browser",
  },
  {
    url: "/dictyaccess",
    description: "dictyAccess",
  },
  {
    url: "/conferences",
    description: "Conference",
  },
  {
    url: "/labs",
    description: "Labs",
  },
  {
    url: "/about",
    description: "About",
  },
  {
    url: "/contact",
    description: "Contact",
  },
]

const theme = {
  primary: "#004080",
  secondary: "#ebe07a",
  text: "#d8d8d8",
}

;<Footer links={links} theme={theme} />
```
