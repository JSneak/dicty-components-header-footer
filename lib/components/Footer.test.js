import React from "react"
import { shallow, configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Footer from "./Footer"
import items from "../data/footer"
import Grid from "@material-ui/core/Grid"

configure({ adapter: new Adapter() })

describe("Footer", () => {
  const wrapper = shallow(<Footer items={items} />).dive()

  describe("initial render", () => {
    it("renders without crashing", () => {
      expect(wrapper).toHaveLength(1)
    })
    it("renders correct number of Grids based on items prop", () => {
      expect(wrapper.find(Grid)).toHaveLength(9)
    })
  })
})
