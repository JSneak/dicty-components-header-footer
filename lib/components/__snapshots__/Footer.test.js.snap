// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Footer 1`] = `
<div
  className="MuiGrid-container-8 MuiGrid-justify-xs-center-25 Footer-root-1"
>
  <div
    className="MuiGrid-item-9 Footer-gridItem-7"
  >
    <div
      className="Footer-subsectionItem-4"
    >
      <div>
        <ul
          className="Footer-headerLink-6"
        >
          <a
            className="Footer-ulLink-5"
            href="http://yahoo.com"
          >
            Genomes
          </a>
        </ul>
      </div>
      <ul>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://cnn.com"
          >
            Genomes Home
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="MuiGrid-item-9 Footer-gridItem-7"
  >
    <div
      className="Footer-subsectionItem-4"
    >
      <div>
        <ul
          className="Footer-headerLink-6"
        >
          <a
            className="Footer-ulLink-5"
            href="http://dictybase.org/tools"
          >
            Tools
          </a>
        </ul>
      </div>
      <ul>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            Tools home
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            New Genomes browser
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="MuiGrid-item-9 Footer-gridItem-7"
  >
    <div
      className="Footer-subsectionItem-4"
    >
      <div>
        <ul
          className="Footer-headerLink-6"
        >
          <a
            className="Footer-ulLink-5"
            href="http://dictybase.org/tools"
          >
            Research
          </a>
        </ul>
      </div>
      <ul>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            Research home
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            Anatomy ontology
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            Codon bias table
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            Nomenclature Guidelines
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/tools"
          >
            Phenotyping
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="MuiGrid-item-9 Footer-gridItem-7"
  >
    <div
      className="Footer-subsectionItem-4"
    >
      <div>
        <ul
          className="Footer-headerLink-6"
        >
          <a
            className="Footer-ulLink-5"
            href="http://dictybase.org/cite"
          >
            Please Cite
          </a>
        </ul>
      </div>
      <ul>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org"
          >
            dictyBase
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/stock"
          >
            Dicty stock center
          </a>
        </li>
      </ul>
    </div>
    <div
      className="Footer-subsectionItem-4"
    >
      <div>
        <ul
          className="Footer-headerLink-6"
        >
          <a
            className="Footer-ulLink-5"
            href="http://dictybase.org/support"
          >
            Supported By
          </a>
        </ul>
      </div>
      <ul>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/nih"
          >
            NIH
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/nih"
          >
            GMOD
          </a>
        </li>
        <li
          className="Footer-listItem-2"
        >
          <a
            className="Footer-link-3"
            href="http://dictybase.org/nih"
          >
            Gene ontology
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
